replica/table.cc

```diff
@@ -3557,8 +3588,25 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
     }
 
     tlogger.info("Cleaned up tablet {} of table {}.{} successfully.", tid, _schema->ks_name(), _schema->cf_name());
+}
+
+future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
+    auto holder = async_gate().hold();
+
+    rwlock::holder exclusive_lock = co_await _sg_manager->get_rwlock().hold_write_lock();
+    std::unique_ptr<storage_group> sg_ptr = co_await _sg_manager->remove_storage_group(exclusive_lock, tid.value());
```

Comment:
No need to pass the exclusive_lock to remove_storage_group. If you only need the lock to be taken, you can grab the lock inside remove_storage_group.

Reply:
Makes sense, since the sg_ptr returned by remove_storage_group is already detached from _storage_groups and the lock is not required to process it further here and in cleanup_tablet_helper.

Reply:
But compaction groups aren't detached. They are kept in boost intrusive list and so during `cleanup_tablet_helper` they are still there